src/basic/path-util.c

```diff
@@ -1441,3 +1441,36 @@ int path_glob_can_match(const char *pattern, const char *prefix, char **ret) {
                 *ret = NULL;
         return false;
 }
+
+const char* default_path(void) {
+#if HAVE_SPLIT_BIN
+        static int split = -1;
+        int r;
+
+        /* Check whether /usr/sbin is separate and return the appropriate $PATH.
+         * On error assume we need split bin as configured… */
+
+        if (split < 0)
+                FOREACH_STRING(path, "/usr/sbin", "/usr/local/sbin") {
+                        _cleanup_free_ char *target = NULL;
+
+                        r = chase(path, /* root= */ NULL, /* flags= */ 0, &target, NULL);
+                        if (r == -ENOENT)
+                                continue;
+                        if (r >= 0 &&
+                            path_equal_or_inode_same(target,
+                                                     streq(path, "/usr/sbin") ? "/usr/bin" : "/usr/local/bin",
```

Comment:
I prefer `path_startswith(path, "/usr/local/")` here. While we only have two fixed cases, let's make the intention a bit more clear to readers.

Reply:
It's doing a literal comparison of the iterator variable with a fixed item being iterated over. And `streq` is the "comparison operator" for strings. So I think it's better here than something else that would give the equivalent result but is actually a more complex operation.